### PR TITLE
Allow explicitly setting decorations to false, closes #133

### LINKED ISSUE #133: Add a way to set decorations to false
Something like `<italic:false>` for example.

### PR DESCRIPTION
**Pull request by MiniDigger**

### CODE REVIEW

**Comment by MiniDigger** [comment]
don't merge before a docs PR has been created...

**Review by zml2008 — DISMISSED** [review]
can you modify the serializer to emit `:false` decoration tags as well? that would make components more roundtrippable :)

**Comment by kezz** [comment]
Does this handle closing as well? For example, would `<italic:false>Hello! <italic>spooky</italic> not spooky</italic:false>` work correctly as well? If so, might be worth throwing a test case for this in.

**Review by kashike — APPROVED** [review]

**Comment by MiniDigger** [comment]
should be good to go, tried my best with the serializer but it just needs a complete rewrite

File: src/main/java/net/kyori/adventure/text/minimessage/parser/node/TagPart.java [hunk 1]
```diff
@@ -127,6 +127,26 @@ public TagPart(
...
   }
 
+  /**
+   * Checks if this tag part represents <code>true</code>.
+   *
+   * @return if this tag part represents <code>true</code>
+   * @since 4.2.0
+   */
+  public boolean isTrue() {
+    return "t".equals(this.value) || "true".equals(this.value) || "on".equals(this.value);
```
Comment (kashike) [inline src/main/java/net/kyori/adventure/text/minimessage/parser/node/TagPart.java:137]: not a fan of `t`

File: src/main/java/net/kyori/adventure/text/minimessage/parser/node/TagPart.java [hunk 2]
```diff
@@ -127,6 +127,26 @@ public TagPart(
...
+  }
+
+  /**
+   * Checks if this tag part represents <code>false</code>.
+   *
+   * @return if this tag part represents <code>false</code>
+   * @since 4.2.0
+   */
+  public boolean isFalse() {
+    return "f".equals(this.value) || "false".equals(this.value) || "off".equals(this.value);
```
Comment (kashike) [inline src/main/java/net/kyori/adventure/text/minimessage/parser/node/TagPart.java:147]: not a fan of `f`

**Review by kezz — APPROVED** [review]
Serializer changes are a bit messy, but not an issue at all considering it'll be rewritten soon.

File: src/main/java/net/kyori/adventure/text/minimessage/transformation/inbuild/DecorationTransformation.java [hunk 3]
```diff
@@ -73,28 +77,39 @@
    * @return a new transformation
    * @since 4.2.0
    */
-  public static DecorationTransformation create(final String name, final List<TagPart> args) {
+  public static DecorationTransformation create(String name, final List<TagPart> args) {
+    boolean flag = args.size() != 1 || !args.get(0).isFalse();
```
Comment (kezz) [inline src/main/java/net/kyori/adventure/text/minimessage/transformation/inbuild/DecorationTransformation.java:81]: Do we want to throw an error for bad arguments? For example, `<b:hello>` would be okay here.

**Pull request merged**